pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -204,25 +204,23 @@ func NewContainerManager(mountUtil mount.Interface, cadvisorInterface cadvisor.I
 		return nil, fmt.Errorf("failed to get mounted cgroup subsystems: %v", err)
 	}
 
-	if failSwapOn {
-		// Check whether swap is enabled. The Kubelet does not support running with swap enabled.
-		swapFile := "/proc/swaps"
-		swapData, err := os.ReadFile(swapFile)
-		if err != nil {
-			if os.IsNotExist(err) {
-				klog.InfoS("File does not exist, assuming that swap is disabled", "path", swapFile)
-			} else {
-				return nil, err
-			}
-		} else {
-			swapData = bytes.TrimSpace(swapData) // extra trailing \n
-			swapLines := strings.Split(string(swapData), "\n")
+	swap.InitializeRootDir(nodeConfig.KubeletRootDir)
```

Comment:
I'm not a fan of having to set a global for IsTmpfsNoswapOptionSupported to work properly

can `IsTmpfsNoswapOptionSupported` take a writeable tmp dir to use? we could feed it here from nodeConfig.KubeletRootDir and from setupTmpfs with `ed.plugin.host.GetPluginDir(emptyDirPluginName)` or something similar

Reply:
Done!

BTW, the reasoning for what I did is that the new `mountPath` parameter added to `IsTmpfsNoswapOptionSupported` will practically be used only at the first use, which will always be called from `container_manager_linux.go` with `KubeletRootDir`. In other words, `ed.plugin.host.GetPluginDir(emptyDirPluginName)` will never be practically used for mounting.

That being said, I think your concern is valid. Code can change and move around and we better ensure it's always safe to use.

Thanks @liggitt! PTAL